libbeat/docs/shared-autodiscover.asciidoc

```diff
@@ -221,18 +221,22 @@ These are the fields available within config templating. The `kubernetes.*` fiel
   * kubernetes.container.image
   * kubernetes.container.name
   * kubernetes.namespace
+  * kubernetes.namespace_uid
```

Comment:
```suggestion
  * kubernetes.namespace.uid
```

Reply:
The field name is indeed kubernetes.namespace_uid.  
@tetianakravchenko maybe we should find a way in this document to add also `kubernetes.labels`, `kubernetes.annotations`, `kubernetes.namespace_labels` and `kubernetes.namespace_annotations`.  Those metadata fields are also part of the metadata fields that can be used in templating when the right parameters are set.  
There is this section in the doc that states 
```
If the `include_annotations` config is added to the provider config, then the list of annotations present in the config
are added to the event.
```
which is ok as an information but we don't list the field. Maybe we could list all fields above and leave those comments there as is.
Also we could add some comments about the namespace_labels and namespace_annotations based on the add_resource_metadata configuration.

Reply:
@gsantoro there is no such field as `kubernetes.namespace.uid` :

<img width="1962" alt="Screenshot 2024-01-10 at 14 18 30" src="https://github.com/elastic/beats/assets/28299531/3e65abdd-4fce-4163-bb49-97f73a9e1d62">

I see that we have such field documented for elastic-agent - https://www.elastic.co/guide/en/fleet/current/kubernetes-provider.html, I believe it is a mistake in the elastic-agent documentation

Reply:
@MichaelKatsoulis I am trying to wrap my head around the docs and the implementation.

I can found two references kubernetes dynamic variables at https://www.elastic.co/guide/en/fleet/current/kubernetes-provider.html and https://www.elastic.co/guide/en/beats/metricbeat/current/configuration-autodiscover.html. For example kubernetes.namespace.uid is in the first one and not in the second doc.

Should those two docs have the same variables and both of them have kubernetes.namespace_uid?

what about this code from elastic-agent-autodiscover where the docs point at namespace.uid and the test at namespace_uid?

I could use some clarification about where in the code those variables are created.

Reply:
@gsantoro kubernetes.namespace is unfortunately a keyword and not an object. There was a try in 8.0 to do this breaking change and put all namespace related fields under the kubernetes.namespace object but it was reverted before it reached production. This was done mainly because of the possible problems it would create to existing customers and the not easy migration procedure.

The documentation in https://www.elastic.co/guide/en/fleet/current/kubernetes-provider.html is wrong. The part of the code where those values are added is here https://github.com/elastic/elastic-agent-autodiscover/blob/b42be0df903aba6580b80761af6df86fa80c845d/kubernetes/metadata/namespace.go#L117